dask/dataframe/categorical.py

```diff
@@ -58,30 +74,63 @@ def categorize(df, columns=None, index=None, **kwargs):
         Whether to categorize the index. By default, object indices are
         converted to categorical, and unknown categorical indices are made
         known. Set True to always categorize the index, False to never.
+    split_every : int, optional
+        Group partitions into groups of this size while performing a
+        tree-reduction. If set to False, no tree-reduction will be used.
+        Default is False.
```

Comment:
Is there a sensible default here?  If we wanted to be conservative this could even be something moderately large like 32.  Having default like this can be helpful when dealing with large clusters.  I've run into cases on 1000-worker clusters where even something very small like `len` is better off using split_every.